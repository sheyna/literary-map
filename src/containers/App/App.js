import styles from './App.module.css';
import SiteHeader from '../../components/SiteHeader/SiteHeader';
import SiteFooter from '../../components/SiteFooter/SiteFooter';
import EnglandMap from '../../views/EnglandMap/EnglandMap';

function App() {

  return (
    <div className={styles.App}>
      <SiteHeader />
      <main className={styles.mainBody}>
      	<EnglandMap />
      </main>
      <SiteFooter />
    </div>
  );

}

export default App;
