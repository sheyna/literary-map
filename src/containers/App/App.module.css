body, html {
	min-height: 100%;
}

.App {
	margin: 0;
	display: flex;
	flex-direction: column;
	min-height: 100%;
}

.mainBody {
  display: flex;
  flex-direction: column;
  flex: 2 0 auto;
  overflow: auto;
}

@media screen and (min-width: 1025px) {
  .mainBody {
    flex-direction: row;
  }
}
