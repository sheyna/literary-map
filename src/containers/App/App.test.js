import { render, screen } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <App />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
