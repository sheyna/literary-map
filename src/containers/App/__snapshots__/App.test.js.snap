// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="SiteHeader"
  >
    <div
      className="title"
    >
      <h1>
        Literary Map
      </h1>
      <h1>
        of England
      </h1>
    </div>
  </header>
  <main
    className="mainBody"
  >
    <div
      className="mapElement"
    >
      <div
        className="MapElement"
      >
        <img
          alt="Map of England"
          src="map-of-england.svg"
        />
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "49%",
              "top": "30%",
            }
          }
        >
          <img
            alt="BrontÃ« Country"
            src="img/brontes-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "27%",
              "top": "77%",
            }
          }
        >
          <img
            alt="A Romance of Exmoor"
            src="img/lorna-doone-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "50%",
              "top": "58%",
            }
          }
        >
          <img
            alt="The Bard of Stratford-upon-Avon"
            src="img/shakespeare-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "38%",
              "top": "20%",
            }
          }
        >
          <img
            alt="William Wordsworth's home"
            src="img/wordsworth-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "60%",
              "top": "22%",
            }
          }
        >
          <img
            alt="Count Dracula Arrives in England"
            src="img/dracula-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "70%",
              "top": "72%",
            }
          }
        >
          <img
            alt="London's Writers' Block"
            src="img/cheyne-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "52%",
              "top": "37%",
            }
          }
        >
          <img
            alt="Romance in the English Civil War"
            src="img/peveril-peak-thumb.png"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "46%",
              "top": "40%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "48%",
              "top": "38%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "62%",
              "top": "78%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "49%",
              "top": "55%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "62%",
              "top": "83%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "45%",
              "top": "85%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "17%",
              "top": "87%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "53%",
              "top": "70%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "39%",
              "top": "85%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
        <div
          className="MapLocationPoint"
          onClick={[Function]}
          style={
            Object {
              "left": "75%",
              "top": "80%",
            }
          }
        >
          <i
            className="fas fa-thumbtack"
          />
        </div>
      </div>
    </div>
    <div
      className="pageContents"
    >
      <p
        className="introText"
      >
        For such a small county, England has produced hundreds of the great works of literature. From Latin and Anglo-Norman literatures to the early development of the English language, stage plays and the birth of the novel, England has seen many types of great literature.
      </p>
    </div>
  </main>
  <footer
    className="SiteFooter"
  >
    <p>
      A project by Sheyna Watkins. Map icon images courtey 
      <a
        href="https://www.davidrumsey.com/about/copyright-and-permissions"
        rel="noreferrer"
        target="_blank"
      >
        David Rumsey Map Collection, David Rumsey Map Center, Stanford Libraries
      </a>
    </p>
  </footer>
</div>
`;
